src/d/actor/d_a_player_main.cpp

```diff
@@ -2210,66 +2210,105 @@ int daPy_lk_c::getDirectionFromAngle(s16 angle) {
 
 /* 80108B08-80108B38       .text getDirectionFromShapeAngle__9daPy_lk_cFv */
 int daPy_lk_c::getDirectionFromShapeAngle() {
-    return getDirectionFromAngle(m34E8 - shape_angle.y);
+    return getDirectionFromAngle(mTargetAngleY - shape_angle.y);
 }
 
 /* 80108B38-80108B68       .text getDirectionFromCurrentAngle__9daPy_lk_cFv */
 int daPy_lk_c::getDirectionFromCurrentAngle() {
-    return getDirectionFromAngle(m34E8 - current.angle.y);
+    return getDirectionFromAngle(mTargetAngleY - current.angle.y);
 }
 
 /* 80108B68-80108D80       .text setNormalSpeedF__9daPy_lk_cFffff */
-void daPy_lk_c::setNormalSpeedF(f32 param_1, f32 param_2, f32 param_3, f32 param_4) {
-    s16 uVar2;
-    f32 dVar6;
-    f32 dVar10;
-
-    if (((dComIfGp_event_runCheck()) || (checkHeavyStateOn())) || (checkGrabWear())) {
-        dVar10 = mMaxNormalSpeed * mStickDistance;
-    } else {
-        dVar10 = mStickDistance * (mMaxNormalSpeed * mStickDistance);
-    }
-    if ((mDemo.getDemoMode() == daPy_demo_c::DEMO_UNK0E_e) || (getSlidePolygon() != NULL)) {
-        return;
-    }
-    if (mAcch.ChkWallHit()) {
-        uVar2 = 0;
-        dBgS_AcchCir* pdVar4 = &mAcchCir[0];
-        for (int i = 0; i < 3; i++, pdVar4++) {
-            if (pdVar4->ChkWallHit()) {
-                uVar2 = (current.angle.y + 0x8000) - pdVar4->GetWallAngleY();
-                break;
-            }
-        }
-        if (abs(uVar2) < 0x4000) {
-            dVar10 *= (1.0f - cM_scos(uVar2) * daPy_HIO_basic_c0::m.field_0x14);
-        }
-    }
-    f32 temp_f3;
-    if (dVar10 < mVelocity) {
-        f32 temp_f0 = mVelocity - dVar10;
-        if (temp_f0 > param_3) {
-            temp_f3 = param_3;
-        } else {
-            temp_f3 = temp_f0;
-        }
-        if (temp_f3 < param_4) {
-            temp_f3 = param_4;
-        }
-        param_1 = 0.0f;
-        dVar6 = dVar10;
-    } else {
-        temp_f3 = param_3;
-        dVar6 = 0.0f;
-    }
-    if (!(cM3d_IsZero(param_1))) {
-        mVelocity += param_1;
-        if (mVelocity > dVar10) {
-            mVelocity = dVar10;
-        }
-    } else {
-        cLib_addCalc(&mVelocity, dVar6, param_2, temp_f3, param_4);
-    }
+void daPy_lk_c::setNormalSpeedF(f32 i_acceleration, f32 i_scale, f32 i_maxStep, f32 i_minStep) {
+   f32 desiredSpeed;
+   // Calculate the desired speed based on the stick distance and Link's maximum normal speed
+   if (dComIfGp_event_getMode() != dEvtMode_NONE_e || checkHeavyStateOn() || field_0x2b0 < 0.0f) {
```

Comment:
Previous inline usage was more likely to be real (dComIfGp_event_runCheck, checkGrabWear)